syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -74,7 +76,7 @@ def on_trial_result(
         self,
         trial_id: int,
         config: Dict[str, Any],
-        metric: float,
+        metric: List[float],
```

Comment:
Can we have another class? Otherwise we have lots of tension in our base class, for instance right now, we have `metric`, `metrics`and the type are sometimes `list[float]` and sometimes `Union[list[float], float]`.

I think we should have another class also because the multiobjective case is something that we only want to use *sometimes*. And we should not put the union of things that we need *sometimes* in the base class.